packages/vega-parser/src/parsers/guides/axis-labels.js

```diff
@@ -77,8 +107,18 @@ export default function(spec, config, userEncode, dataRef, size, band) {
     }
   };
 
+  if (!isSignal(orient)) {
```

Comment:
It is confusing to have multiple if blocks here, particularly one that checks `isSignal(orient)` followed by one that checks `!isSignal(orient)`. Can we either consolidate into one if block, or break the branches apart into different helper functions?

Reply:
I've consolidated the evaluation of `labelAlign`, `labelBaseline`, and `offset` into a single block by combining the logic for signal and non-signal cases using a resolving function. Happy to make additional changes here.